Spigot-Server-Patches/0288-Implement-furnace-cook-speed-multiplier-API.patch

```diff
@@ -45,25 +50,39 @@ index 99b20fa5feff0f766124d4ec9474852e33e329f2..1d3c2dd93657fb5dc71ee6b444c585b5
          ContainerUtil.a(nbttagcompound, this.items);
          NBTTagCompound nbttagcompound1 = new NBTTagCompound();
  
-@@ -295,8 +303,8 @@ public abstract class TileEntityFurnace extends TileEntityContainer implements I
-                 }
+@@ -296,7 +304,7 @@ public abstract class TileEntityFurnace extends TileEntityContainer implements I
  
                  if (this.isBurning() && this.canBurn(irecipe)) {
--                    ++this.cookTime;
+                     ++this.cookTime;
 -                    if (this.cookTime == this.cookTimeTotal) {
-+                    this.cookTime += cookSpeedMultiplier; // Paper - cook speed multiplier API
 +                    if (this.cookTime >= this.cookTimeTotal) { // Paper - cook speed multiplier API
                          this.cookTime = 0;
                          this.cookTimeTotal = this.getRecipeCookingTime();
                          this.burn(irecipe);
+@@ -396,9 +404,13 @@ public abstract class TileEntityFurnace extends TileEntityContainer implements I
+         }
+     }
+ 
+-    protected int getRecipeCookingTime() {
+-        return (this.hasWorld()) ? (Integer) this.world.getCraftingManager().craft((Recipes<RecipeCooking>) this.c, this, this.world).map(RecipeCooking::getCookingTime).orElse(200) : 200; // CraftBukkit - SPIGOT-4302 // Eclipse fail
++    // Paper begin - Expose this function so CraftFurnace can correctly scale the total cooking time to a new multiplier
++    public int getRecipeCookingTime() {
++        /* Scale the recipe's cooking time to the current cookSpeedMultiplier */
++        Integer cookTime = (this.hasWorld()) ? (Integer) this.world.getCraftingManager().craft((Recipes<RecipeCooking>) this.c, this, this.world).map(RecipeCooking::getCookingTime).orElse(200) : 200; // CraftBukkit - SPIGOT-4302 // Eclipse fail
```

Comment:
Prefer `int` here; there is no need in using a boxed value here.

Reply:
Just changed it to an `int` in my end! I initially used an `Integer` wrapper because the original code already casted the current recipe's cook time into an `Integer` in the first place.